src/modules/samples/rust/src/lib.rs

```diff
@@ -23,14 +22,46 @@ pub extern "C" fn MmiGetInfo(
     payload: *mut MmiJsonString,
     payload_size_bytes: *mut c_int,
 ) -> c_int {
-    unimplemented!("MmiGetInfo is not yet implemented");
+    if client_name.is_null() {
+        println!("MmiGetInfo called with null clientName");
+        EINVAL
+    } else if payload.is_null() {
+        println!("MmiGetInfo called with null payload");
+        EINVAL
+    } else if payload_size_bytes.is_null() {
+        println!("MmiGetInfo called with null payloadSizeBytes");
+        EINVAL
+    } else {
+        // Borrow the client_name ptr
+        let client_name_cstr: &CStr = unsafe { CStr::from_ptr(client_name) };
+        let client_name_str_slice: &str;
+        match client_name_cstr.to_str() {
```

Comment:
Please try to have a single return point in this function. You can return from a `match` block like how a status code `libc::c_int` is returned from each branch of the if-else statement.

Reply:
clippy and rust-analyzer may have some good suggestions here too for the late initialization of the `&str` declared before each `match` block. Lets discuss